doc/source/debugging.rst

```diff
@@ -0,0 +1,171 @@
+*********************
+Debugging and Logging
+*********************
+
+.. versionchanged:: 1.4.0
+
+   Ophyd's use of Python's logging framework has been completely reworked to
+   follow Python's documented best practices for libraries.
+
+Ophyd uses Python's logging framework, which enables sophisticated log
+management. For common simple cases, including viewing logs in the terminal or
+writing them to a file, the next section illustrates streamlined,
+copy/paste-able examples. Users who are familiar with that framework or who
+need to route logs to multiple destinations may wish to skip ahead to
+:ref:`logger_api`.
+
+Useful Snippets
+===============
+
+Log warnings
+------------
+
+This is the recommended standard setup.
+
+.. code-block:: python
+
+   from ophyd import config_ophyd_logging
+   config_ophyd_logging()
+
+It will display ``'ophyd'`` log records of ``WARNING`` level or higher in the
+terminal (standard out) with a format tailored to ophyd.
+
+Maximum verbosity
+-----------------
+
+If operations are "hanging," running slowly, or repeatedly encountering an
+error, increasing the logging verbosity can help identify the underlying issue.
+
+.. code-block:: python
+
+   from ophyd import config_ophyd_logging
+   config_ophyd_logging(level='DEBUG')
+
+Log to a file
+-------------
+
+This will direct all log messages to a file instead of the terminal (standard
+out).
+
+.. code-block:: python
+
+    from ophyd import config_ophyd_logging
+    config_ophyd_logging(file='/tmp/ophyd.log', level='DEBUG')
+
+.. _logger_api:
+
+Ophyd's Logging-Related API
+=============================
+
+Logger Names
+------------
+
+Here are the primary loggers used by ophyd.
+
+* ``'ophyd'`` --- the logger to which all ophyd log records propagate
+* ``'ophyd.control_layer'`` --- logs requests issued to the underlying control
+  layer (e.g. pyepics, caproto)
+* ``'ophyd.event_dispatcher'`` --- issues regular summaries of the backlog of
```

Comment:
@klauer Is this a precise description? Would you add anything?

Reply:
Seems about right. Considering we're the target audience of the log output, what you have is more than sufficient in my opinion.